# IS 362 Assignment 4: Airports and Weather

## Introduction

This assignment uses Python and pandas to analyze data from the `nycflights13` dataset on GitHub.

The analysis will answer three questions:

1. What is the northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12, 2013, which New York area airport had the windiest weather?

The analysis will use airport information and weather data from the `nycflights13` CSV files. The results will also be researched and verified using additional information where necessary.

In [1]:
import pandas as pd

In [2]:
airports = pd.read_csv("airports.csv")

print (airports.head())

   faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York  


In [3]:
northernmost = airports.sort_values("lat", ascending=False).head(5)

print (northernmost[["faa", "name", "lat", "lon"]])

     faa                                        name        lat         lon
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003
110  AIN                          Wainwright Airport  70.638056 -159.994722
708  K03                               Wainwright As  70.613378 -159.860350
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000


## Question 1: Northernmost Airport

The initial pandas analysis identified Dillant Hopkins Airport (EEN) as the northernmost airport because its latitude was listed as 72.270833 degrees north in the CSV file.

However, this result appears to be an outlier in the source data. Dillant Hopkins Airport is located in New Hampshire, and additional airport information confirms that its latitude is approximately 42.898 degrees north.

Therefore, I will treat the latitude value of 72.270833 for EEN as an apparent data error and correct it before determining the northernmost airport.

In [4]:
airports.loc[airports["faa"] == "EEN", "lat"] = 42.898

In [5]:
northernmost = airports.sort_values("lat", ascending=False).head(5)

print (northernmost[["faa", "name", "lat", "lon"]])

      faa                                        name        lat         lon
230   BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003
110   AIN                          Wainwright Airport  70.638056 -159.994722
708   K03                               Wainwright As  70.613378 -159.860350
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000
1363  UUK                        Ugnu-Kuparuk Airport  70.330833 -149.597500


The corrected data shows that the five northernmost airports in the dataset are all located in Alaska. Wiley Post Will Rogers Memorial Airport (BRW) has the highest latitude at 71.285446 degrees north, followed by Wainwright Airport (AIN), Wainwright AS (K03), Atqasuk Edward Burnell Sr Memorial Airport (ATK), and Ugnu-Kuparuk Airport (UUK).

Based on the corrected dataset, BRW is the northernmost airport among the candidates examined. This result will be verified using additional airport information.

### Verification

I verified the result using additional airport information on Google. Wiley Post-Will Rogers Memorial Airport (BRW) is located in Utqiagvik, Alaska, at approximately 71.29 degrees north latitude. This agrees closely with the latitude of 71.285446 degrees listed in the CSV data.

Therefore, the answer to Question 1 is **Wiley Post-Will Rogers Memorial Airport (BRW)**.

In [6]:
easternmost = airports.sort_values("lon", ascending=False).head(5)

print (easternmost[["faa", "name", "lat", "lon"]])

      faa                           name        lat         lon
1290  SYA                   Eareckson As  52.712275  174.113620
942   MYF               Montgomery Field  32.475900  117.759000
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000
417   EEN        Dillant Hopkins Airport  42.898000   42.898333
444   EPM     Eastport Municipal Airport  44.910111  -67.012694


## Question 2: Easternmost Airport

The initial pandas analysis identified Eareckson Air Station (SYA) as the easternmost airport in the dataset, with a longitude of 174.113620 degrees east.

The other four airports in the five-airport result have longitudes west of Greenwich. The EEN result is an apparent data error because its longitude is listed as 42.898333, which is nearly identical to its corrected latitude.

I will verify the SYA result using additional airport information from Google before giving the final answer to Question 2.

In [7]:
airports[airports["faa"].isin(["SYA", "EPM"])][["faa", "name", "lat", "lon"]]

,faa,name,lat,lon
444,EPM,Eastport Municipal Airport,44.910111,-67.012694
1290,SYA,Eareckson As,52.712275,174.113620


In [8]:
airports[airports["faa"] == "SYA"].T

,1290
faa,SYA
name,Eareckson As
lat,52.712275
lon,174.11362
alt,98
tz,-9
dst,A
tzone,America/Anchorage


### Verification

The pandas analysis identified Eareckson Air Station (SYA) as the easternmost airport when the longitude values were sorted numerically. However, additional research from Google showed that Eastport Municipal Airport (EPM) is identified as the easternmost airport in the United States.

According to information from U.S. Senator Susan Collins' website (https://www.collins.senate.gov/newsroom/senator-collins-participates-eastport-airport-groundbreaking), Eastport Municipal Airport is a city-owned, public-use airport and is described as the easternmost airport in the United States.

Therefore, based on the additional research, the answer to Question 2 is **Eastport Municipal Airport (EPM)**. The difference between the pandas result and the researched answer demonstrates why the airport data needed to be verified using additional information.

In [9]:
weather = pd.read_csv("weather.csv")
print (weather.head())

  origin  year  month  day  hour   temp   dewp  humid  wind_dir  wind_speed  \
0    EWR  2013      1    1     1  39.02  26.06  59.37     270.0    10.35702   
1    EWR  2013      1    1     2  39.02  26.96  61.63     250.0     8.05546   
2    EWR  2013      1    1     3  39.02  28.04  64.43     240.0    11.50780   
3    EWR  2013      1    1     4  39.92  28.04  62.21     250.0    12.65858   
4    EWR  2013      1    1     5  39.02  28.04  64.43     260.0    12.65858   

   wind_gust  precip  pressure  visib             time_hour  
0        NaN     0.0    1012.0   10.0  2013-01-01T06:00:00Z  
1        NaN     0.0    1012.3   10.0  2013-01-01T07:00:00Z  
2        NaN     0.0    1012.5   10.0  2013-01-01T08:00:00Z  
3        NaN     0.0    1012.2   10.0  2013-01-01T09:00:00Z  
4        NaN     0.0    1011.9   10.0  2013-01-01T10:00:00Z  


In [10]:
feb12 = weather[
    (weather["year"] == 2013) &
    (weather["month"] == 2) &
    (weather["day"] == 12)
]

print (feb12[["origin", "hour", "wind_speed", "wind_gust"]])

      origin  hour  wind_speed  wind_gust
1006     EWR     0     6.90468        NaN
1007     EWR     1     9.20624        NaN
1008     EWR     2    20.71404   25.31716
1009     EWR     3  1048.36058        NaN
1010     EWR     4    12.65858        NaN
...      ...   ...         ...        ...
18434    LGA    19    14.96014   21.86482
18435    LGA    20    10.35702        NaN
18436    LGA    21    10.35702        NaN
18437    LGA    22    11.50780        NaN
18438    LGA    23     9.20624        NaN

[72 rows x 4 columns]


In [11]:
windiest = feb12.sort_values("wind_speed", ascending=False)

print (windiest[["origin", "hour", "wind_speed", "wind_gust"]].head(10))

      origin  hour  wind_speed  wind_gust
1009     EWR     3  1048.36058        NaN
18417    LGA     2    23.01560   31.07106
1018     EWR    12    21.86482   31.07106
18428    LGA    13    21.86482   25.31716
9717     JFK     8    20.71404   27.61872
1008     EWR     2    20.71404   25.31716
18429    LGA    14    20.71404   25.31716
18416    LGA     1    19.56326   28.76950
9716     JFK     7    19.56326        NaN
18427    LGA    12    19.56326   26.46794


In [12]:
ewr_feb12 = feb12[feb12["origin"] == "EWR"]

print (ewr_feb12[["hour", "wind_dir", "wind_speed", "wind_gust"]])

      hour  wind_dir  wind_speed  wind_gust
1006     0     240.0     6.90468        NaN
1007     1     250.0     9.20624        NaN
1008     2     270.0    20.71404   25.31716
1009     3     260.0  1048.36058        NaN
1010     4     280.0    12.65858        NaN
1011     5     270.0    13.80936        NaN
1012     6     270.0    16.11092        NaN
1013     7     260.0    14.96014   20.71404
1014     8     250.0    17.26170        NaN
1015     9     290.0    12.65858        NaN
1016    10     260.0    17.26170   23.01560
1017    11     260.0    17.26170        NaN
1018    12     270.0    21.86482   31.07106
1019    13     280.0    13.80936   31.07106
1020    14     280.0    17.26170   24.16638
1021    15     300.0    12.65858   19.56326
1022    16     270.0    13.80936        NaN
1023    17     260.0    16.11092   26.46794
1024    18     270.0    11.50780        NaN
1025    19     270.0    12.65858        NaN
1026    20     280.0     9.20624        NaN
1027    21     300.0     5.75390

In [13]:
print (1048.36058 / 1.15078)

911.0000000000001


In [14]:
feb12_clean = feb12[feb12["wind_speed"] < 100].copy()

print (feb12_clean[["origin", "hour", "wind_speed", "wind_gust"]].sort_values("wind_speed", ascending=False).head(10))

      origin  hour  wind_speed  wind_gust
18417    LGA     2    23.01560   31.07106
1018     EWR    12    21.86482   31.07106
18428    LGA    13    21.86482   25.31716
9717     JFK     8    20.71404   27.61872
1008     EWR     2    20.71404   25.31716
18429    LGA    14    20.71404   25.31716
18416    LGA     1    19.56326   28.76950
9716     JFK     7    19.56326        NaN
18427    LGA    12    19.56326   26.46794
18430    LGA    15    17.26170   20.71404


In [15]:
max_wind_by_airport = feb12_clean.groupby("origin")["wind_speed"].max()

print (max_wind_by_airport)

origin
EWR    21.86482
JFK    20.71404
LGA    23.01560
Name: wind_speed, dtype: float64


### Handling the Outlier

The initial analysis showed an unusually high wind speed of 1048.36058 mph at EWR at hour 3. This value was far outside the range of the other observations for the day.

I investigated the value by dividing it by 1.15078, the approximate conversion factor from knots to miles per hour. The result was approximately 911, suggesting that the value may have originated from a raw value of 911 that was converted to miles per hour. Because 1048.36058 mph is not a plausible weather observation, I treated this value as an outlier and excluded it from the comparison rather than replacing it with an invented value.

After excluding the outlier, the maximum wind speeds were:

- EWR: 21.86482 mph
- JFK: 20.71404 mph
- LGA: 23.01560 mph

Therefore, LaGuardia Airport (LGA) had the highest recorded wind speed among the three New York area airports on February 12, 2013.

### Answer to Question 3

After accounting for the outlier in the weather data, **LaGuardia Airport (LGA)** had the windiest weather among the three New York area airports on February 12, 2013.

Its highest recorded wind speed was **23.01560 mph**, compared with **21.86482 mph at EWR** and **20.71404 mph at JFK**.

Therefore, the answer to Question 3 is **LaGuardia Airport (LGA)**.